src/server_manager/web_app/digitalocean_app/create_server_app.ts

```diff
@@ -0,0 +1,128 @@
+import {Location, OutlineRegionPicker} from '../ui_components/outline-region-picker-step';
+import {XhrError} from "../../cloud/digitalocean_api";
+
+// DigitalOcean mapping of regions to flags
+const FLAG_IMAGE_DIR = 'images/flags';
+const DIGITALOCEAN_FLAG_MAPPING: {[cityId: string]: string} = {
+  ams: `${FLAG_IMAGE_DIR}/netherlands.png`,
+  sgp: `${FLAG_IMAGE_DIR}/singapore.png`,
+  blr: `${FLAG_IMAGE_DIR}/india.png`,
+  fra: `${FLAG_IMAGE_DIR}/germany.png`,
+  lon: `${FLAG_IMAGE_DIR}/uk.png`,
+  sfo: `${FLAG_IMAGE_DIR}/us.png`,
+  tor: `${FLAG_IMAGE_DIR}/canada.png`,
+  nyc: `${FLAG_IMAGE_DIR}/us.png`,
+};
+
+@customElement('digitalocean-create-server-app')
+export class DigitalOceanCreateServerApp extends LitElement {
+  public static EVENT_SERVER_CREATED = 'do-create-server-app-server-created';
+  public static EVENT_AUTHORIZATION_ERROR = 'do-create-server-app-authorization-error';
+
+  @property({type: Function}) localize: Function;
+
+  private digitalOceanServerRepository: ManagedServerRepository;
+  private notificationManager: OutlineNotificationManager;
+
+  render() {
+    return html`<outline-region-picker-step .localize=${this.localize} 
+        @outline-region-picker-region-selected="${
+        this.onRegionSelected}"></outline-region-picker-step>`;
+  }
+
+  async onRegionSelected(event: CustomEvent) {
+    const regionPicker =
+        this.shadowRoot.querySelector('outline-region-picker-step') as OutlineRegionPicker;
+    regionPicker.isServerBeingCreated = true;
+
+    try {
+      const server =
+          await this.createServer(this.digitalOceanServerRepository, event.detail.regionId);
+      const serverCreatedEvent =
+          makePublicEvent(DigitalOceanCreateServerApp.EVENT_SERVER_CREATED, {server});
+      this.dispatchEvent(serverCreatedEvent);
+    } catch (error) {
+      this.notificationManager.showError('error-server-creation');
+    }
+  }
+
+  setNotificationManager(notificationManager: OutlineNotificationManager) {
+    this.notificationManager = notificationManager;
+  }
+
+  async start(digitalOceanServerRepository: ManagedServerRepository): Promise<void> {
+    this.digitalOceanServerRepository = digitalOceanServerRepository;
+    return this.showRegionPicker(this.digitalOceanServerRepository);
+  }
+
+  private async showRegionPicker(digitalOceanServerRepository: ManagedServerRepository):
+      Promise<void> {
+    const regionPicker =
+        this.shadowRoot.querySelector('outline-region-picker-step') as OutlineRegionPicker;
+    regionPicker.isServerBeingCreated = false;
+    regionPicker.reset();
+
+    try {
+      // TODO: Catch and rethrow authorization error
+      const regionMap = await digitalOceanServerRepository.getRegionMap();
+      const locations = Object.entries(regionMap).map(([cityId, regionIds]) => {
+        return this.createLocationModel(cityId, regionIds);
+      });
+      regionPicker.locations = locations;
+    } catch (error) {
+      if (error instanceof XhrError) {
```

Comment:
This doesn't read right. Why is a XhrError an authorization error? Could you make this more readable?

Also, don't you have to check the status code?